moveit_experimental/moveit_jog_arm/src/jog_calcs.cpp

```diff
@@ -124,25 +140,19 @@ void JogCalcs::startMainLoop(JogArmShared& shared_variables, std::mutex& mutex)
   ros::Rate loop_rate(1. / parameters_.publish_period);
 
   // Now do jogging calcs
-  while (ros::ok() && !stop_requested_)
+  while (ros::ok())
   {
-    // Flag that incoming commands are all zero. May be used to skip calculations/publication
-    mutex.lock();
-    bool zero_cartesian_cmd_flag = shared_variables.zero_cartesian_cmd_flag;
-    bool zero_joint_cmd_flag = shared_variables.zero_joint_cmd_flag;
-    mutex.unlock();
-
-    if (zero_cartesian_cmd_flag && zero_joint_cmd_flag)
-      // Reset low-pass filters
-      resetVelocityFilters();
-
-    // Pull data from the shared variables.
+    // Always update the joints and end-effector transform for 2 reasons:
+    // 1) in case the getCommandFrameTransform() method is being used
+    // 2) so the low-pass filters are up to date and don't cause a jump
     mutex.lock();
     incoming_joints_ = shared_variables.joints;
     mutex.unlock();
-
+    while (!updateJoints() && ros::ok() && !stop_requested_)
+    {
+      ros::Duration(WHILE_LOOP_WAIT).sleep();
```

Comment:
Maybe unrelated to this PR, but is there a reason we are not using `ros::Rate`?

Reply:
Not that I know of.  ;)  Will update it

Reply:
I found out there is a reason. The default sleep rate is defined in jog_arm_data.h:

`static const double WHILE_LOOP_WAIT = 0.001;`

If I change that to `static ros::Rate DEFAULT_SLEEP_RATE(1000)'` the error is:

Cannot use ros::Time::now() before the first NodeHandle has been created or ros::start() has been called

Reply:
Yes, ros::Rate can't be static, you'll have to initialize it before the loop (or inside constructor). A big advantage is that ros::Rate will run the loop in the rate you specified while `ros::Duration::sleep()` adds a constant sleep additional to the computation time.

Reply:
ok, done